Firmware/MotorControl/motor.cpp

```diff
@@ -440,13 +440,14 @@ bool Motor::FOC_current(float Id_des, float Iq_des, float I_phase, float pwm_pha
 }
 
 
-bool Motor::update(float current_setpoint, float phase, float phase_vel) {
+bool Motor::update(float torque_setpoint, float phase, float phase_vel) {
+    float current_setpoint = torque_setpoint / config_.torque_constant;
     current_setpoint *= config_.direction;
     phase *= config_.direction;
     phase_vel *= config_.direction;
 
     // TODO: 2-norm vs independent clamping (current could be sqrt(2) bigger)
-    float ilim = effective_current_lim();
+    float ilim = effective_torque_lim() / config_.torque_constant;
     float id = std::clamp(current_control_.Id_setpoint, -ilim, ilim);
     float iq = std::clamp(current_setpoint, -ilim, ilim);
```

Comment:
Note that `iq = torque_setpoint / torque_constant` doesn't work for ACIM motors. In that case you need to do:
`iq = torque_setpoint / (torque_constant * acim_rotor_flux)`. Beware of dividing by zero acim_rotor_flux, you can use `max(acim_rotor_flux, acim_gain_min_flux)`